### PR TITLE
Adding label filtering

### PR DESCRIPTION
**Pull request by pubudu538**
### Proposed changes in this pull request

- Added filtering mechanism for labels when retrieving. 

### When should this PR be merged

ASAP.

### Follow up actions

N/A

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] ** Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] ** Are all UI and API inputs run through forms or serializers?**
- [ ] ** Are all external inputs validated and sanitized appropriately?**
- [ ] ** Does all branching logic have a default case?**
- [ ] ** Does this solution handle outliers and edge cases gracefully?**
- [ ] ** Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java [hunk 1]
```diff
@@ -440,15 +445,34 @@ public Policy getPolicy(String policyLevel, String policyName) throws APIManagem
...
-        List<Label> labelList;
+        List<Label> filteredLabels;
+        String labelExtractorClassName = config.getLabelExtractor();
         try {
-            labelList = getLabelDAO().getLabelsByName(labels);
+            List<Label> availableLabels = getLabelDAO().getLabelsByName(labels);
+
+            // TODO: Need to update this based on the roles impl.
+            // Logged in user
+            String user = "admin";
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java]: Is there any special reason to hard-code user here.

Comment (Rajith90) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java]: Can't we use getUsername() method in the AbstractAPIManager class

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/exception/ExceptionCodes.java [hunk 2]
```diff
@@ -94,7 +94,8 @@
             " Non existing roles cannot be added to an API"),
     LABEL_INFORMATION_CANNOT_BE_NULL(900331, "Label information cannot be null", 400, "Label information cannot be " +
             "null"),
-    APPLICATION_INACTIVE(900332, "Application is not active", 400, "Application is not active");
+    APPLICATION_INACTIVE(900332, "Application is not active", 400, "Application is not active"),
+    LABEL_INFORMATION_EXCEPTION(900332, "Label Info error", 500, "Error occurred while retrieving label information");
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/exception/ExceptionCodes.java]: same exception codes. please increment the value

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java (hunk 1 above)
Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java]: Used hard-coded values as we have hard coded users at the moment. Based on a discussion, this implementation has changed and roles will not be used. Username of the logged in user will be used.

**Pull request merged**